src/callback_requests.rs

```diff
@@ -208,8 +203,21 @@ pub enum CallbackRequestType {
         since: Instant,
     },
 
+    /// Check if the user can permissions to perform some operations
     KubernetesCanI {
-        subject_access_review: Box<SubjectAccessReviewWrapper>,
+        /// Describe the set of parameters used by the `can_i` function. The values in this struct
+        /// will be used to build the SubjectAccessReview resources sent to the Kubernetes API to
+        /// verify if the user is allowed to perform some operation
+        request: KWSubjectAccessReviewRequest,
+
+        /// Disable caching of results obtained from Kubernetes API Server
+        /// By default query results are cached for 5 seconds, that might cause
+        /// stale data to be returned.
+        /// However, making too many requests against the Kubernetes API Server
+        /// might cause issues to the cluster
+        ///
+        /// This field is also defined in the `KWSubjectAccessReviewRequest`. However, we duplicate
+        /// the value here to keep the same pattern used by the other Kubernetes requests
         disable_cache: bool,
```

Comment:
Maybe I'm missing something, but I don't see why there's a need to allow for this disable_cache. This means that we expect/send 2 `disable_cache` keys in the payload. IIUC only the one inside the KWSubjectAccessReviewRequest is honoured.

This is also contrary to what we do for GetResourceRequest too, as those payloads only have 1 `disable_cache`.

Reply:
@flavio @viccuad ,  let's decide what we want to do about this. Do you want to move the `disable_cache` to another struct and add the current `SubjectAccessReviewRequest` into it together? Something like:

```rust
struct CanIRequest {
  subject_access_review_data: SubjectAccessReviewRequest,
  disable_cache: bool
}
```

Reply:
@viccuad Sorry, I just saw your proposal [here](https://github.com/kubewarden/policy-sdk-rust/pull/187#discussion_r2200277738). Which are the same thing that I just wrote here as well. I like the idea we both came up with. I vote for it as well.